plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -451,9 +604,46 @@ public function get_description(): string {
 	 * @param WP_Post $post The deleted post.
 	 */
 	private function handle_deleted_post( $postid, $post ): void {
-		if ( 'shop_order' === $post->post_type && $this->data_sync_is_enabled() ) {
+		global $wpdb;
+
+		if ( 'shop_order' !== $post->post_type && 'shop_order_refund' !== $post->post_type ) {
+			return;
+		}
+
+		$features_controller = wc_get_container()->get( FeaturesController::class );
+		$feature_is_enabled  = $features_controller->feature_is_enabled( 'custom_order_tables' );
+		if ( ! $feature_is_enabled ) {
+			return;
+		}
+
+		if ( $this->data_sync_is_enabled() ) {
 			$this->data_store->delete_order_data_from_custom_order_tables( $postid );
+		} elseif ( $this->custom_orders_table_is_authoritative() ) {
+			return;
+		}
+
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared, WordPress.DB.SlowDBQuery
+		if ( $wpdb->get_var(
+			$wpdb->prepare(
+				"SELECT EXISTS (SELECT id FROM {$this->data_store::get_orders_table_name()} WHERE ID=%d)
+						AND NOT EXISTS (SELECT order_id FROM {$this->data_store::get_meta_table_name()} WHERE order_id=%d AND meta_key=%s AND meta_value=%s)",
+				$postid,
+				$postid,
+				'deleted_from',
+				$wpdb->posts
+			)
+		)
+		) {
+			$wpdb->insert(
+				$this->data_store::get_meta_table_name(),
+				array(
+					'order_id'   => $postid,
+					'meta_key'   => 'deleted_from',
```

Comment:
should `deleted_from` be a constant defined somewhere? perhaps it should also start from `_` to mark it internal.

Reply:
Implemented in https://github.com/woocommerce/woocommerce/pull/37050/commits/e83eb705999548066ee8ad72c3d9abc53af13e3c